src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -202,8 +205,21 @@ public function supportsDenormalization($data, $type, $format = null, array $con
      */
     private function getNormalizer($data, $format, array $context)
     {
+        if (\is_object($data) && isset($this->normalizerCache[\get_class($data)][$format])) {
+            return $this->normalizerCache[\get_class($data)][$format];
```

Comment:
There will be some BC break here, let's imagine two normalizers with same class and format where:

 * The first one (having a high priority) vary on data so not cacheable
 * The second one (having a lower priorirty) vary on class only, so is cacheable

When a data will get the second one, then it will have more priority than the first one (as it will be cached and returned here everytime).

Reply:
No because we don't cache anything if one of the normalizer in the chain before the one that marches is not cacheable (even if the normalizer finally used is cacheable).